### ISSUE TITLE
Wurst Randomly Crashes When I tap the control Button bug

### ISSUE THREAD

**Issue by cleveravi** [Wurst-Imperium/Wurst-Client#1668]
When I have wurst active and I try to hit the control button wurst just randomly crashes and says this.

```
---------------- Wurst Client Crash Report ----------------
---- Please report this crash at: wurst-client.tk/bugs ----
// My bad.

Time: 8/5/15 4:02 PM
Description: Rendering screen

java.lang.ArrayIndexOutOfBoundsException: -1
	at org.darkstorm.minecraft.gui.component.basic.BasicComboBox.getSelectedElement(BasicComboBox.java:65)
	at org.darkstorm.minecraft.gui.theme.wurst.WurstComboBoxUI.renderComponent(WurstComboBoxUI.java:230)
	at org.darkstorm.minecraft.gui.theme.wurst.WurstComboBoxUI.renderComponent(WurstComboBoxUI.java:1)
	at org.darkstorm.minecraft.gui.theme.AbstractComponentUI.render(AbstractComponentUI.java:32)
	at org.darkstorm.minecraft.gui.component.AbstractComponent.render(AbstractComponent.java:32)
	at org.darkstorm.minecraft.gui.component.AbstractContainer.render(AbstractContainer.java:28)
	at org.darkstorm.minecraft.gui.component.basic.BasicFrame.render(BasicFrame.java:35)
	at org.darkstorm.minecraft.gui.AbstractGuiManager.render(AbstractGuiManager.java:112)
	at org.darkstorm.minecraft.gui.util.GuiManagerDisplayScreen.drawScreen(GuiManagerDisplayScreen.java:119)
	at net.minecraft.client.renderer.EntityRenderer.updateCameraAndRender(EntityRenderer.java:1180)
	at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1298)
	at net.minecraft.client.Minecraft.run(Minecraft.java:383)
	at net.minecraft.client.main.Main.main(Main.java:127)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at org.darkstorm.minecraft.gui.component.basic.BasicComboBox.getSelectedElement(BasicComboBox.java:65)
	at org.darkstorm.minecraft.gui.theme.wurst.WurstComboBoxUI.renderComponent(WurstComboBoxUI.java:230)
	at org.darkstorm.minecraft.gui.theme.wurst.WurstComboBoxUI.renderComponent(WurstComboBoxUI.java:1)
	at org.darkstorm.minecraft.gui.theme.AbstractComponentUI.render(AbstractComponentUI.java:32)
	at org.darkstorm.minecraft.gui.component.AbstractComponent.render(AbstractComponent.java:32)
	at org.darkstorm.minecraft.gui.component.AbstractContainer.render(AbstractContainer.java:28)
	at org.darkstorm.minecraft.gui.component.basic.BasicFrame.render(BasicFrame.java:35)
	at org.darkstorm.minecraft.gui.AbstractGuiManager.render(AbstractGuiManager.java:112)
	at org.darkstorm.minecraft.gui.util.GuiManagerDisplayScreen.drawScreen(GuiManagerDisplayScreen.java:119)

-- Screen render details --
Details:
	Screen name: org.darkstorm.minecraft.gui.util.GuiManagerDisplayScreen
	Mouse location: Scaled: (261, 84). Absolute: (522, 544)
	Screen size: Scaled: (640, 357). Absolute: (1280, 714). Scale factor of 2

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [EntityPlayerSP['cleveravi'/321, l='MpServer', x=4.02, y=62.00, z=262.68]]
	Chunk stats: MultiplayerChunkCache: 169, 169
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: 4.00,64.00,256.00 - World: (4,64,256), Chunk: (at 4,4,0 in 0,16; contains blocks 0,0,256 to 15,255,271), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 130795 game time, 130795 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
	Forced entities: 71 total; [EntityItem['item.item.beefRaw'/131, l='MpServer', x=-19.16, y=72.00, z=243.81], EntityItem['item.item.beefRaw'/132, l='MpServer', x=-18.03, y=73.00, z=245.22], EntityItem['item.item.leather'/133, l='MpServer', x=-26.91, y=73.00, z=251.44], EntityItem['item.item.leather'/261, l='MpServer', x=68.13, y=68.00, z=219.91], EntityItem['item.item.leather'/134, l='MpServer', x=-18.00, y=73.00, z=244.81], EntityItem['item.item.saddle'/262, l='MpServer', x=67.88, y=67.00, z=220.78], EntityItem['item.item.leather'/135, l='MpServer', x=-16.72, y=77.00, z=290.34], EntityItem['item.item.horsearmordiamond'/263, l='MpServer', x=67.00, y=67.00, z=221.19], EntityZombie['Zombie'/264, l='MpServer', x=64.50, y=13.00, z=247.50], EntityZombie['Zombie'/265, l='MpServer', x=64.50, y=13.00, z=246.50], EntitySkeleton['Skeleton'/266, l='MpServer', x=74.45, y=20.47, z=291.58], EntityCreeper['Creeper'/267, l='MpServer', x=73.50, y=21.00, z=292.50], EntityItem['item.item.arrow'/145, l='MpServer', x=-13.25, y=26.00, z=212.91], EntityItem['item.item.bone'/146, l='MpServer', x=-13.13, y=26.00, z=212.84], EntityItem['item.item.arrow'/147, l='MpServer', x=-10.78, y=25.00, z=210.13], EntityItem['item.item.bone'/148, l='MpServer', x=-11.88, y=26.00, z=211.00], EntityItem['item.item.porkchopRaw'/149, l='MpServer', x=-15.06, y=70.00, z=220.78], EntitySkeleton['Skeleton'/150, l='MpServer', x=-9.50, y=42.00, z=291.50], EntityItem['item.item.horsearmordiamond'/151, l='MpServer', x=-15.31, y=76.00, z=290.75], EntityItem['item.item.saddle'/152, l='MpServer', x=-15.13, y=76.00, z=291.25], EntityItem['item.item.rottenFlesh'/153, l='MpServer', x=-2.75, y=43.00, z=308.88], EntityItem['item.item.arrow'/170, l='MpServer', x=7.13, y=40.00, z=269.56], EntityItem['item.item.arrow'/171, l='MpServer', x=7.91, y=40.00, z=269.75], EntityItem['item.item.bone'/172, l='MpServer', x=8.31, y=40.00, z=269.97], EntityCreeper['Creeper'/173, l='MpServer', x=15.50, y=39.00, z=302.50], EntityCreeper['Creeper'/174, l='MpServer', x=14.03, y=36.00, z=294.44], EntitySkeleton['Skeleton'/175, l='MpServer', x=15.50, y=38.00, z=301.50], EntityItem['item.item.rottenFlesh'/176, l='MpServer', x=9.88, y=35.00, z=291.78], EntityItem['item.item.rottenFlesh'/177, l='MpServer', x=8.88, y=35.00, z=291.13], EntityItem['item.item.arrow'/178, l='MpServer', x=13.88, y=35.00, z=294.13], EntityItem['item.item.bone'/179, l='MpServer', x=13.22, y=35.00, z=294.13], EntityEnderman['Enderman'/180, l='MpServer', x=15.50, y=35.00, z=290.50], EntityPlayerSP['cleveravi'/321, l='MpServer', x=4.02, y=62.00, z=262.68], EntityCreeper['Creeper'/209, l='MpServer', x=22.64, y=19.76, z=204.37], EntityItem['item.item.rottenFlesh'/210, l='MpServer', x=19.69, y=19.00, z=214.06], EntitySkeleton['Skeleton'/211, l='MpServer', x=18.50, y=20.00, z=213.50], EntityItem['item.item.leather'/212, l='MpServer', x=24.47, y=67.00, z=270.13], EntityItem['item.item.saddle'/213, l='MpServer', x=24.94, y=67.00, z=270.25], EntityItem['item.item.saddle'/214, l='MpServer', x=30.81, y=67.00, z=263.59], EntityItem['item.item.sulphur'/215, l='MpServer', x=22.13, y=43.00, z=289.81], EntityRabbit['Rabbit'/97, l='MpServer', x=-72.50, y=63.00, z=193.50], EntityPig['Pig'/98, l='MpServer', x=-74.13, y=63.00, z=196.47], EntityZombie['Zombie'/99, l='MpServer', x=-71.94, y=35.00, z=239.78], EntityBat['Bat'/105, l='MpServer', x=-61.53, y=26.71, z=335.02], EntitySkeleton['Skeleton'/109, l='MpServer', x=-50.50, y=26.00, z=238.50], EntityItem['item.item.porkchopRaw'/110, l='MpServer', x=-55.66, y=78.00, z=274.38], EntityBat['Bat'/111, l='MpServer', x=-59.85, y=24.91, z=333.51], EntitySkeleton['Skeleton'/240, l='MpServer', x=42.69, y=13.03, z=235.28], EntityZombie['Zombie'/112, l='MpServer', x=-60.50, y=25.00, z=331.50], EntityZombie['Zombie'/241, l='MpServer', x=36.50, y=43.10, z=232.95], EntityBat['Bat'/113, l='MpServer', x=-54.86, y=24.09, z=326.81], EntitySpider['Spider'/242, l='MpServer', x=37.94, y=42.00, z=230.97], EntityItem['item.item.sulphur'/243, l='MpServer', x=38.31, y=15.00, z=241.78], EntityItem['item.item.leather'/244, l='MpServer', x=34.53, y=69.00, z=247.66], EntityZombie['Zombie'/245, l='MpServer', x=45.91, y=55.00, z=261.59], EntityZombie['Zombie'/246, l='MpServer', x=43.50, y=55.00, z=261.50], EntityZombie['Zombie'/247, l='MpServer', x=35.50, y=52.00, z=268.50], EntityItem['item.item.sulphur'/248, l='MpServer', x=45.13, y=27.00, z=283.00], EntityCreeper['Creeper'/120, l='MpServer', x=-39.80, y=19.00, z=191.96], EntityItem['item.item.arrow'/249, l='MpServer', x=41.59, y=27.00, z=283.69], EntitySpider['Spider'/121, l='MpServer', x=-41.50, y=19.00, z=191.50], EntityZombie['Zombie'/250, l='MpServer', x=37.47, y=37.00, z=284.63], EntitySpider['Spider'/122, l='MpServer', x=-38.50, y=19.00, z=193.50], EntityZombie['Zombie'/251, l='MpServer', x=32.50, y=39.00, z=282.50], EntitySpider['Spider'/123, l='MpServer', x=-36.50, y=19.00, z=192.50], EntityItem['item.item.rottenFlesh'/252, l='MpServer', x=47.16, y=13.00, z=299.09], EntityBat['Bat'/124, l='MpServer', x=-40.50, y=26.47, z=204.97], EntityCreeper['Creeper'/125, l='MpServer', x=-45.50, y=28.00, z=211.50], EntityItem['item.item.sulphur'/254, l='MpServer', x=55.13, y=42.00, z=299.84], EntityZombie['Zombie'/126, l='MpServer', x=-44.50, y=17.00, z=342.50], EntityItem['item.item.sulphur'/255, l='MpServer', x=55.13, y=42.00, z=299.03]]
	Retry entities: 0 total; []
	Server brand: vanilla
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.multiplayer.WorldClient.addWorldInfoToCrashReport(WorldClient.java:393)
	at net.minecraft.client.Minecraft.addGraphicsAndWorldToCrashReport(Minecraft.java:2925)
	at net.minecraft.client.Minecraft.run(Minecraft.java:405)
	at net.minecraft.client.main.Main.main(Main.java:127)

-- System Details --
Details:
	Minecraft Version: 1.8
	Wurst Client Version: 2.1 (latest: 2.1)
	Operating System: Mac OS X (x86_64) version 10.9.5
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 361510624 bytes (344 MB) / 684056576 bytes (652 MB) up to 1060372480 bytes (1011 MB)
	JVM Flags: 5 total; -Xmx1G -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn128M
	IntCache: cache: 0, tcache: 0, allocated: 13, tallocated: 95
	Launched Version: Wurst
	LWJGL: 2.9.2
	OpenGL: Intel HD Graphics 4000 OpenGL Engine GL version 2.1 INTEL-8.28.34, Intel Inc.
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because ARB_framebuffer_object is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Very likely; Jar signature invalidated
	Type: Client (map_client.txt)
	Resource Packs: [Wolion128_3D]
	Current Language: English (US)
	Profiler Position: N/A (disabled)
```

Please Help I've opened this about and hour ago

**Comment by OMGchangeMYusername** [comment]
They will help you once they're on, I just don't know what's goin' on :P

**Comment by cleveravi** [comment]
Thanks

**Comment by GummiBoat** [comment]
# *Duplicate: #640*

![](https://cloud.githubusercontent.com/assets/10976441/9021444/990e8990-3842-11e5-809e-a06b796a4328.png)

For the future, please consider putting crashes into a [code-markdown block](https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks).

**Comment by OMGchangeMYusername** [comment]
um.. How do you find crash duplicates?

**Comment by GummiBoat** [comment]
With magic.

**Comment by cleveravi** [comment]
It still is not working
---------------- Wurst Client Crash Report ----------------
---- Please report this crash at: wurst-client.tk/bugs ----
// Uh... Did I do that?

Time: 8/7/15 10:48 AM
Description: Rendering screen

java.lang.ArrayIndexOutOfBoundsException: -1
	at org.darkstorm.minecraft.gui.component.basic.BasicComboBox.getSelectedElement(BasicComboBox.java:65)
	at org.darkstorm.minecraft.gui.theme.wurst.WurstComboBoxUI.renderComponent(WurstComboBoxUI.java:230)
	at org.darkstorm.minecraft.gui.theme.wurst.WurstComboBoxUI.renderComponent(WurstComboBoxUI.java:1)
	at org.darkstorm.minecraft.gui.theme.AbstractComponentUI.render(AbstractComponentUI.java:32)
	at org.darkstorm.minecraft.gui.component.AbstractComponent.render(AbstractComponent.java:32)
	at org.darkstorm.minecraft.gui.component.AbstractContainer.render(AbstractContainer.java:28)
	at org.darkstorm.minecraft.gui.component.basic.BasicFrame.render(BasicFrame.java:35)
	at org.darkstorm.minecraft.gui.AbstractGuiManager.render(AbstractGuiManager.java:112)
	at org.darkstorm.minecraft.gui.util.GuiManagerDisplayScreen.drawScreen(GuiManagerDisplayScreen.java:119)
	at net.minecraft.client.renderer.EntityRenderer.updateCameraAndRender(EntityRenderer.java:1180)
	at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1298)
	at net.minecraft.client.Minecraft.run(Minecraft.java:383)
	at net.minecraft.client.main.Main.main(Main.java:127)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at org.darkstorm.minecraft.gui.component.basic.BasicComboBox.getSelectedElement(BasicComboBox.java:65)
	at org.darkstorm.minecraft.gui.theme.wurst.WurstComboBoxUI.renderComponent(WurstComboBoxUI.java:230)
	at org.darkstorm.minecraft.gui.theme.wurst.WurstComboBoxUI.renderComponent(WurstComboBoxUI.java:1)
	at org.darkstorm.minecraft.gui.theme.AbstractComponentUI.render(AbstractComponentUI.java:32)
	at org.darkstorm.minecraft.gui.component.AbstractComponent.render(AbstractComponent.java:32)
	at org.darkstorm.minecraft.gui.component.AbstractContainer.render(AbstractContainer.java:28)
	at org.darkstorm.minecraft.gui.component.basic.BasicFrame.render(BasicFrame.java:35)
	at org.darkstorm.minecraft.gui.AbstractGuiManager.render(AbstractGuiManager.java:112)
	at org.darkstorm.minecraft.gui.util.GuiManagerDisplayScreen.drawScreen(GuiManagerDisplayScreen.java:119)

-- Screen render details --
Details:
	Screen name: org.darkstorm.minecraft.gui.util.GuiManagerDisplayScreen
	Mouse location: Scaled: (348, 44). Absolute: (348, 435)
	Screen size: Scaled: (854, 480). Absolute: (854, 480). Scale factor of 1

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [EntityPlayerSP['cleveravi'/315, l='MpServer', x=5.48, y=63.00, z=263.32]]
	Chunk stats: MultiplayerChunkCache: 169, 169
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: 4.00,64.00,256.00 - World: (4,64,256), Chunk: (at 4,4,0 in 0,16; contains blocks 0,0,256 to 15,255,271), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 131691 game time, 131691 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
	Forced entities: 33 total; [EntityZombie['Zombie'/258, l='MpServer', x=69.13, y=12.00, z=244.63], EntityZombie['Zombie'/259, l='MpServer', x=64.41, y=13.00, z=245.75], EntitySkeleton['Skeleton'/260, l='MpServer', x=74.63, y=21.00, z=291.22], EntityCreeper['Creeper'/261, l='MpServer', x=73.50, y=21.00, z=292.50], EntitySkeleton['Skeleton'/147, l='MpServer', x=-11.53, y=42.00, z=293.09], EntityCreeper['Creeper'/171, l='MpServer', x=8.59, y=40.00, z=303.47], EntitySkeleton['Skeleton'/172, l='MpServer', x=13.56, y=37.00, z=300.69], EntityPlayerSP['cleveravi'/315, l='MpServer', x=5.48, y=63.00, z=263.32], EntitySkeleton['Skeleton'/204, l='MpServer', x=29.25, y=17.00, z=202.22], EntityCreeper['Creeper'/205, l='MpServer', x=29.03, y=17.00, z=201.34], EntityEnderman['Enderman'/210, l='MpServer', x=16.59, y=34.00, z=289.63], EntityCreeper['Creeper'/211, l='MpServer', x=20.06, y=36.00, z=288.50], EntitySkeleton['Skeleton'/214, l='MpServer', x=17.53, y=48.00, z=341.34], EntityPig['Pig'/96, l='MpServer', x=-72.34, y=63.00, z=194.09], EntityRabbit['Rabbit'/97, l='MpServer', x=-68.19, y=63.00, z=194.50], EntityZombie['Zombie'/98, l='MpServer', x=-71.94, y=35.00, z=239.78], EntitySkeleton['Skeleton'/234, l='MpServer', x=42.69, y=13.06, z=235.28], EntitySkeleton['Skeleton'/106, l='MpServer', x=-50.50, y=26.00, z=238.50], EntityZombie['Zombie'/235, l='MpServer', x=38.38, y=42.00, z=232.06], EntitySpider['Spider'/236, l='MpServer', x=35.69, y=44.09, z=227.69], EntityZombie['Zombie'/109, l='MpServer', x=-59.47, y=26.00, z=330.94], EntityZombie['Zombie'/239, l='MpServer', x=42.50, y=55.00, z=262.94], EntityZombie['Zombie'/240, l='MpServer', x=52.44, y=63.14, z=262.35], EntityZombie['Zombie'/241, l='MpServer', x=35.50, y=52.00, z=268.50], EntityZombie['Zombie'/244, l='MpServer', x=45.53, y=28.00, z=280.47], EntityCreeper['Creeper'/116, l='MpServer', x=-37.50, y=19.00, z=189.50], EntityZombie['Zombie'/245, l='MpServer', x=32.50, y=39.00, z=282.50], EntitySpider['Spider'/117, l='MpServer', x=-41.50, y=19.00, z=191.50], EntitySpider['Spider'/118, l='MpServer', x=-38.50, y=19.00, z=193.50], EntitySpider['Spider'/119, l='MpServer', x=-36.50, y=19.00, z=192.50], EntityBat['Bat'/120, l='MpServer', x=-45.25, y=27.10, z=208.75], EntityCreeper['Creeper'/121, l='MpServer', x=-45.50, y=28.00, z=211.50], EntityZombie['Zombie'/123, l='MpServer', x=-42.72, y=17.00, z=342.50]]
	Retry entities: 0 total; []
	Server brand: vanilla
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.multiplayer.WorldClient.addWorldInfoToCrashReport(WorldClient.java:393)
	at net.minecraft.client.Minecraft.addGraphicsAndWorldToCrashReport(Minecraft.java:2925)
	at net.minecraft.client.Minecraft.run(Minecraft.java:405)
	at net.minecraft.client.main.Main.main(Main.java:127)

-- System Details --
Details:
	Minecraft Version: 1.8
	Wurst Client Version: 2.1 (latest: 2.1)
	Operating System: Mac OS X (x86_64) version 10.9.5
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 328913656 bytes (313 MB) / 683896832 bytes (652 MB) up to 1060372480 bytes (1011 MB)
	JVM Flags: 5 total; -Xmx1G -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn128M
	IntCache: cache: 0, tcache: 0, allocated: 13, tallocated: 95
	Launched Version: Wurst
	LWJGL: 2.9.2
	OpenGL: Intel HD Graphics 4000 OpenGL Engine GL version 2.1 INTEL-8.28.34, Intel Inc.
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because ARB_framebuffer_object is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Very likely; Jar signature invalidated
	Type: Client (map_client.txt)
	Resource Packs: [Wolion128_3D]
	Current Language: English (US)
	Profiler Position: N/A (disabled)
